AustinHarris.JsonRpcTestN/Test.cs

```diff
@@ -1879,6 +1879,13 @@ public void TestNestedReturnType()
             Assert.AreEqual(expected, result.Result);
         }
 
+        [Test]
+        public void InstantiatingAServiceMoreThanOnceDoesNotThrow()
+        {
+            Assert.DoesNotThrow(()=>new CalculatorService());
+            Assert.DoesNotThrow(() => new CalculatorService());
```

Comment:
I think those _workarounds_ are fantastic! We should add them to the getting started documentation. I currently believe that it is the expected and correct behavior to throw the exception when trying to register a handler for the same call twice in the same scope. We could possible make it a little nicer by having additional ways to register services that will only do it if it's needed, or let you provide a delegate to create the service.

The reason I am still feeling this way is because throwing the exception makes it very clear to the developer that we could not register their handler and that it failed.

Based off of this test (https://github.com/Astn/JSON-RPC.NET/blob/master/AustinHarris.JsonRpcTestN/Test.cs#L63)
See this contrived scenario:
```
        [Test()]
        public void TestCanCreateAndRemoveSession()
        {
            var h = JsonRpc.Handler.GetSessionHandler("this one");
            var metadata = new System.Collections.Generic.List<Tuple<string, Type>> {
                Tuple.Create ("fruit", typeof(string)),
                Tuple.Create ("returns", typeof(string))
            }.ToDictionary(x => x.Item1, x => x.Item2);

// First Registration - A function that returns "Red - $fruit" is bound to the "getColor" function
            h.RegisterFuction("getColor", metadata, new System.Collections.Generic.Dictionary<string, object>(),new Func<string, string>(x => "Red -" + x));

// Duplicate Registration (Throws Error under current code) - A function that returns "Blue - $fruit" is bound to the "getColor" function. This will never fire if "getColor" is already bound (which it is). We currently throw an error under this condition. Your patch would prevent the error, but the test would still fail. Maybe we should make the error message to be better? If we swallow the error, then the developer will not know that this binding failed.
            h.RegisterFuction("getColor", metadata, new System.Collections.Generic.Dictionary<string, object>(),new Func<string, string>(x => "Blue -" + x));

            string request = @"{method:'getColor',params:{'fruit':'Bananna'},id:1}";
            string expectedResult = "{\"jsonrpc\":\"2.0\",\"result\":\"Red - Bananna\",\"id\":1}";
            string expectedResultAfterDestroy = "{\"jsonrpc\":\"2.0\",\"error\":{\"message\":\"Method not found\",\"code\":-32601,\"data\":\"The method does not exist / is not available.\"},\"id\":1}";
            var result = JsonRpcProcessor.Process("this one", request);
            result.Wait();

            var actual1 = JObject.Parse(result.Result);
            var expected1 = JObject.Parse(expectedResult);
            Assert.AreEqual(expected1, actual1);

            h.Destroy();

            var result2 = JsonRpcProcessor.Process("this one", request);
            result2.Wait();

            Assert.AreEqual(JObject.Parse(expectedResultAfterDestroy), JObject.Parse(result2.Result));
        }
```

What do you think?